src/Appwrite/Platform/Workers/Webhooks.php

```diff
@@ -117,10 +133,87 @@ private function execute(array $events, string $payload, Document $webhook, Docu
+
+            if ($attempts >= self::MAX_FAILED_ATTEMPTS) {
+                $webhook->setAttribute('enabled', false);
+
+                $memberships = $dbForConsole->find('memberships', [
+                    Query::equal('teamInternalId', [$project->getAttribute('teamInternalId')]),
+                    Query::limit(APP_LIMIT_SUBQUERY)
+                ]);
+
+                $userIds = array_column(\array_map(fn ($membership) => $membership->getArrayCopy(), $memberships), 'userId');
+
+                $users = $dbForConsole->find('users', [
+                    Query::equal('$id', $userIds),
+                ]);
+
+                $protocol = App::getEnv('_APP_OPTIONS_FORCE_HTTPS') == 'disabled' ? 'http' : 'https';
+                $hostname = App::getEnv('_APP_DOMAIN');
+                $projectId = $project->getId();
+                $webhookId = $webhook->getId();
+
+                $template = Template::fromFile(__DIR__ . '/../../../../app/config/locale/templates/email-webhook-failed.tpl');
+
+                $template->setParam('{{webhook}}', $webhook->getAttribute('name'));
+                $template->setParam('{{project}}', $project->getAttribute('name'));
+                $template->setParam('{{url}}', $webhook->getAttribute('url'));
+                $template->setParam('{{error}}', $curlError ??  'The server returned ' . $statusCode . ' status code');
+                $template->setParam('{{redirect}}', $protocol . '://' . $hostname . "/console/project-$projectId/settings/webhooks/$webhookId");
+                $template->setParam('{{attempts}}', $attempts);
+
+                $subject = 'Webhook deliveries have been paused';
+                $body = Template::fromFile(__DIR__ . '/../../../../app/config/locale/templates/email-base-styled.tpl');
+
+                $body
+                    ->setParam('{{subject}}', $subject)
+                    ->setParam('{{message}}', $template->render())
+                    ->setParam('{{year}}', date("Y"));
+
+                $queueForMails
+                    ->setSubject($subject)
+                    ->setBody($body->render());
+
+                foreach ($users as $user) {
+                    $queueForMails
+                        ->setVariables(['user' => $user->getAttribute('name', '')])
+                        ->setName($user->getAttribute('name', ''))
+                        ->setRecipient($user->getAttribute('email'))
+                        ->trigger();
+                }
+            }
+
+            $dbForConsole->updateDocument('webhooks', $webhook->getId(), $webhook);
+            $dbForConsole->deleteCachedDocument('projects', $project->getId());
+
+            $this->errors[] = $logs;
+        } else {
+            $webhook->setAttribute('attempts', 0); // Reset attempts on success
+            $dbForConsole->updateDocument('webhooks', $webhook->getId(), $webhook);
+            $dbForConsole->deleteCachedDocument('projects', $project->getId());
```

Comment:
you should also reset the logs attribute here 👍

Reply:
@eldadfux mentioned that we should not reset the logs, otherwise, the user will be confused where did the logs suddenly disappear. Let me know whatever is a better product decision.